Terminal.Gui/ConsoleDrivers/V2/ApplicationV2.cs

```diff
@@ -0,0 +1,218 @@
+﻿#nullable enable
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.Logging;
+
+namespace Terminal.Gui;
+
+/// <summary>
+///     Implementation of <see cref="IApplication"/> that boots the new 'v2'
+///     main loop architecture.
+/// </summary>
+public class ApplicationV2 : ApplicationImpl
+{
+    private readonly Func<INetInput> _netInputFactory;
```

Comment:
Not sure I like having Application be so tightly bound to "net" and "win". What about curses/linux? Or is the idea "net" works great on linux?

Reply:
Yes my current thinking is that if we can get away with only the two drivers then it will make things simpler.

I think core dotnet console apis will work fast and well enough now on Linux for prime time - but we can test and see if there are distros that have performance issues that would need curses.

Application is the ugliest bit of this PR as it is where the old architectural approach (static gateway + driver monolith) meets the new (orchestrator + generics subcomponents).

I think these factories and how we boot strap new architecture will evolve here.

This needs to be good enough not perfect (my hope anyway).